src/System.Diagnostics.EventLog/tests/EventLogTests/EventLogSourceCreationTests.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System.Diagnostics;
+using Xunit;
+using System;
+
+namespace System.Diagnostics.Tests
+{
+    public class EventLogSourceCreationTests : EventLogTestsBase
+    {
+        [ConditionalFact(nameof(IsProcessElevated))]
+        public void CheckSourceExistanceAndDeletion()
+        {
+            string source = Guid.NewGuid().ToString("N");
+            if (!EventLog.SourceExists(source))
+            {
+                EventLog.CreateEventSource(source, "MyNewLog");
+            }
+            Assert.True(EventLog.SourceExists(source));
+            EventLog.DeleteEventSource(source);
+            Assert.False(EventLog.SourceExists(source));
+
+        }
+
+        [ConditionalFact(nameof(IsProcessElevated))]
+        public void CheckSourceExistsArgumentNull()
+        {
+            Assert.False(EventLog.SourceExists(null));
+        }
+
+        [ConditionalFact(nameof(IsProcessElevated))]
+        public void DeleteUnregisteredSource()
+        {
+            Assert.Throws<System.ArgumentException>(() => EventLog.DeleteEventSource(Guid.NewGuid().ToString("N")));
```

Comment:
nit: `Assert.Throws<ArgumentException>` and also add the parameter overload to check the thrown argument name. If these are different between Desktop and Core use AssertExtensions instead.

Reply:
Both with overloading and using assertExtentions ..both are giving param string as null which should be "source"